[PFSP1-0001][PFSP1-0002] Initial Screen, Login Screen

evan03121998 opened the pull request:
Closes #20
Closes #21 
***
## Description
Initial Screen View

## Related Links
List related PRs against other branches:

List | link
------ | ------
User Story | [link]()
Figma | [link]()
Issue | [link]()

## Todos
- [X] UI
- [ ] Coordinator
- [ ] Network
- [ ] Tests
- [ ] Documentation

## Screenshot / Video
Screenshot consisting of changes made on this branch.

### UI Changes
UI related changes will be shown here (can be either screenshot or video format)

### Network Status
Network related changes will be shown here (screenshot of request status)

### Test Coverage
Test coverage result for designated changes will be shown here (screenshot of coverage)

## Notes
Additional notes that you have either technical debt, blocker, etc.

evan03121998 committed:
Initial Screen View

inxudianz commented on MainApp/MainApp/Feature/Splash/InitialScreenViewController.swift:
```diff
@@ -0,0 +1,33 @@
+    }
+
+
+    /*
+    // MARK: - Navigation
+
+    // In a storyboard-based application, you will often want to do a little preparation before navigation
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        // Get the new view controller using segue.destination.
+        // Pass the selected object to the new view controller.
+    }
+    */
```
comment2 dihapus bang

evan03121998 replied:
g

inxudianz commented on MainApp/MainApp/Feature/Splash/InitialScreenViewController.swift:
```diff
@@ -0,0 +1,33 @@
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Do any additional setup after loading the view.
```
commend dihapus

evan03121998 replied:
g

inxudianz commented on MainApp/MainApp/Feature/Splash/InitialScreenViewController.swift:
```diff
@@ -0,0 +1,33 @@
+
+    @IBOutlet weak var initialScreenImageView: UIImageView!
+    @IBOutlet weak var initialScreenLabel: UILabel!
+    @IBOutlet weak var initialScreenArrowButton: UIButton!
```
buat IBAction untuk navigate to next screen sama buat coordinator

evan03121998 committed:
Updates :
IBAction to go to login page
Coordinator

inxudianz reviewed (changes requested):
Space - space yang berlebihan dihapus

Coordinator related file taro di Coordinator/Feature group

evan03121998 committed:
Updates :
Move Coordinator and CoordinatorProtocol files to Coordinator group
Delete excess spacing

evan03121998 committed:
Merge branch 'develop' into feature/PFSP1-0001

evan03121998 committed:
login UI
login coordinator

inxudianz reviewed (changes requested):
Login coordinator ga ke commit

inxudianz commented on MainApp/MainApp/Feature/Login/LoginViewController.swift:
```diff
@@ -0,0 +1,42 @@
+    }
+
+    @IBAction func loginTapped(_ sender: Any) {
+        viewModel?.gotoHome()
```
logic login email sama gitu2 nya di handle juga ya, tutorialnya entar besok sore

evan03121998 committed:
LoginCoordinator file

evan03121998 committed:
coordinator from Initial Screen to Login func gotoLogin()

inxudianz committed:
Merge branch 'develop' into feature/PFSP1-0001

evan03121998 committed:
update xib login button

evan03121998 committed:
LoginModel
LoginResponse
LoginPath
LoginSubPath
LoginService

evan03121998 committed:
LoginNetwork

evan03121998 committed:
LoginService

inxudianz commented on MainApp/MainApp/FeatureNetwork/SampleNetwork/LoginNetwork.swift:
```diff
@@ -0,0 +1,45 @@
+//  Created by Evan Christian on 06/08/20.
+//  Copyright © 2020 William Inx. All rights reserved.
+//
+
```
Pindahin LoginNetwork di group LoginNetwork

inxudianz commented on MainApp/MainApp/FeatureNetwork/SampleNetwork/LoginNetwork.swift:
```diff
@@ -0,0 +1,45 @@
+//
+
+import Foundation
+import RxSwift
```
RXSwift no need to be imported

inxudianz commented on MainApp/MainApp/Network/Service/LoginService.swift:
```diff
@@ -0,0 +1,51 @@
+    var task: HTTPTask {
+        switch self {
+        case .loginRequest:
+            return .plainRequest
```
login karena ada pake model pakenya parameteredRequest

inxudianz commented on MainApp/MainApp/Network/Path/FeaturePath/Sample/LoginPath.swift:
```diff
@@ -0,0 +1,13 @@
+//  LoginPath.swift
+//  MainApp
+//
+//  Created by Evan Christian on 06/08/20.
```
pindahin ke FeaturePath/Auth/<File>

inxudianz commented on MainApp/MainApp/Network/Path/FeaturePath/Sample/LoginSubPath.swift:
```diff
@@ -0,0 +1,13 @@
+//  Created by Evan Christian on 06/08/20.
+//  Copyright © 2020 William Inx. All rights reserved.
+//
+
```
pindahin ke FeaturePath/Auth/<File>

evan03121998 committed:
requests done
